lib/internal/worker/js_transferable.js

```diff
@@ -22,21 +24,29 @@ function setup() {
     const { 0: module, 1: ctor } = StringPrototypeSplit(deserializeInfo, ':');
     const Ctor = require(module)[ctor];
     if (typeof Ctor !== 'function' ||
-        !(Ctor.prototype instanceof JSTransferable)) {
+        typeof Ctor.prototype[messaging_deserialize_symbol] !== 'function') {
       // Not one of the official errors because one should not be able to get
       // here without messing with Node.js internals.
       // eslint-disable-next-line no-restricted-syntax
       throw new Error(`Unknown deserialize spec ${deserializeInfo}`);
     }
+
     return new Ctor();
   });
 }
 
+function makeTransferable(obj) {
+  const inst = ReflectConstruct(JSTransferable, [], obj.constructor);
+  ObjectSetPrototypeOf(inst, obj);
```

Comment:
```suggestion
  ObjectDefineProperties(inst, ObjectGetOwnPropertyDescriptors(obj));
  ObjectSetPrototype(inst, ObjectGetPrototype(obj));
```

Otherwise own properties of `obj` will not be own properties of `inst`, and `obj` would be a mutable prototype object which is usually bad for performance.